walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,52 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Long duration = request.getDuration();
+    Iterator<Event> iterator = events.iterator();
+    int start = 0;
+    Collection<TimeRange> timeBlock = new ArrayList();
+    if(duration>TimeRange.WHOLE_DAY.duration()){
+        return Arrays.asList();
+    }
+    if(events.size()==0 ){
```

Comment:
Please ensure the spacing on your if statements stay consistent like so:

`if (events.size()==0) {`